rfcs/0000-wifi-buddy-interface.md

```diff
@@ -0,0 +1,125 @@
+- Information about the network interface, such as signal strength
+- Errors and other diagnostic information reported by the WiFi Buddy
+- `trace-proto`-formatted logs from the WiFi buddy (?)
+- Potentially, messages related to Bluetooth Low Energy operation (out of scope
+  for this document)
+
+It may also be desirable for the CPU to eventually be able to update the
+firmware on the WiFI buddy. However, for now, both of the target WiFi Buddy
+devices can be flashed over USB, so the MnemOS CPU being able to update the WiFi
+Buddy firmware is not a high priority at this time.
+
+Data path communication will be performed over the SPI bus, as it's the
+highest-bandwidth link available with this hardware. In [#121], we have
+tentatively concluded that the firmware will use a "dumb WiFi Buddy" design,
+where TCP or any other higher-level network protocols are implemented on the
+CPU, and the CPU and WiFi Buddy communicate by exchanging MAC-layer [802.3]
+Ethernet frames.
+
+## Design Questions
+
+Based on the background information discussed above, we can begin to enumerate
+design questions for the WiFi Buddy interface:
+
+- **Is control path communication in-band or out of band?** If data path frames
+  are exchanged between the host CPU and the WiFi Buddy over the SPI bus, do we
+  also exchange control messages on the same bus? Or, are control messages sent
+  using another interface, such as I²C?
+
+  Using a single interface for both control and data path communication is
+  conceptually simpler, and has the advantage of a clear and obvious ordering
+  between data and control messages. It also means that we don't necessarily
+  need to implement drivers for all the interfaces available on the ESP32-C3.
+
+  Downsides of in-band control messages include that we must introduce some
+  additional form of tagging of data sent on the SPI bus, in order  to indicate
+  what bytes are part of a control message and what bytes are actual network
+  data. This may introduce some additional overhead. On the other hand, if we
+  want to eventually include Bluetooth Low Energy as well as WiFi, we will need
+  the ability to indicate what data is an Ethernet frame and what is BLE,
+  anyway, so we'll already be introducing some overhead for tagging data.
+
+- **Do we want the CPU to be able to receive `trace-proto` from WiFi Buddy?**
+  Both the CPU and WiFi Buddy devices will emit debugging information using
+  [`mnemos-trace-proto`]. Do we want `trace-proto`-formatted logs from the WiFi
+  Buddy to be exposed to the CPU, so that it can proxy them to an attached debug
+  host? Or, is using the WiFi Buddy SoM's onboard USB-serial hardware sufficient
+  for debugging purposes?
+
+- **What control messages are necessary?** Eventually, we'll need to enumerate
+  what control messages will need to be exchanged in order to implement the
+  required functionality.
+
+- **How does the WiFi Buddy signal that data is ready?** Since the CPU is the
```

Comment:
Definitely +1 that we need at least ONE GPIO signal line, even if it is optional. Possible useful lines are:

* "Modem ready to receive data" (can also indicate "I am busy processing" or "buffers are full")
* "Modem has data for host"